accepted/00063-hub-iss.md

```diff
@@ -0,0 +1,246 @@
+  - be limited to software channels
+  - be limited to storage-based transfers
+
+## Implementation details
+
+- two commandline tools will be added, one to be used on an ISS Master and one on an ISS Slave
+  - output of the ISS Master tool will be a directory with files and a SQL script
+  - the ISS Slave tool will mainly copy files to correct locations and execute the SQL script
+  - triggering of the ISS Slave tool might happen via Salt, this is left as an implementation detail
+  - transfer proper could be done via `rsync`, which provides transparent compression and delta encoding
+- all new code will be typed Python 3, connecting to Postgres via psycopg2 and no dependencies to the old Python stack
+- inspection of the schema can be implemented via `SELECT` queries on Postgres's internal tables
+  - current and new implementations would thus be completely independent, so could coexist for an initial period of time, as the new implementation matures
+
+## Impact on existing components and users
+All code would be new in a new component, so no change to existing components is expected. No impact on existing users is expected at all until the old implementation is dropped.
+
+When that happens, there is a chance of regressions for ISS users, which represent a small minority of total users.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Syncing content between different versions of Uyuni on the ISS Master and on the ISS Slave may not work, if the database schema changes in non-compatible ways.
+
+This risk is fundamentally unavoidable at a conceptual level, although different mitigations can be put in place and the current implementation actually allows for a broader set of mitigations than the proposal in this RFC can allow.
+
+In particular:
+
+1. Some changes in the database structure could be "internal only", not really changing the nature of the information that's being exchanged. In those cases, the current implementation could make use of the fact the the XML representation does not change, thus the ISS Slave continues to operate normally
+2. Other changes would be reflected in the XML structure, but still not create problems because, for example, the current implementation ignores unknown tags. This is the case for completely new elements that the ISS Slave would just not import (which is most likely conceptually correct, as tables would also be missing)
+3. Finally, yet other changes could really introduce a backwards compatibility problem and, since the protocol is versioned, [the import program will send an alert and abort](https://github.com/SUSE/spacewalk/blob/Manager-3.2/backend/satellite_tools/xmlSource.py#L271-L298).
+
+With the proposal in this RFC, it is possible to implement protection from case 3. via a similar versioning mechanism, partially against 2. ignoring missing tables but only offer limited resilience against case 1., as we would not have a real database-agnostic representation of data.
```

Comment:
I think this is the key point and we need to think carefully about it. Is it acceptable that we force a customer to update there whole Hub/Server infrastructure "together" at the same day? In worst case we talk about 1 Hub with 9k Servers.

Otherwise it he is not able to sync content from hub to not migrated servers.

Another point is the "internal only" problem. Are all developers aware that they potentially break ISS when they restruture DB tables and data? It should not be too easy to break it.